lib/crowdsec.lua

```diff
@@ -490,15 +608,18 @@ function csmod.Allow(ip)
               ngx.log(ngx.ERR, "Error while validating captcha: " .. err)
             end
             if valid == true then
-                -- captcha is valid, we redirect the IP to its previous URI but in GET method
-                local succ, err, forcible = ngx.shared.crowdsec_cache:set("captcha_"..ngx.var.remote_addr, previous_uri, runtime.conf["CAPTCHA_EXPIRATION"], captcha.GetStateID(captcha._VALIDATED_STATE))
-                if not succ then
-                  ngx.log(ngx.ERR, "failed to add key about captcha for ip '" .. ngx.var.remote_addr .. "' in cache: "..err)
-                end
-                if forcible then
-                  ngx.log(ngx.ERR, "Lua shared dict (crowdsec cache) is full, please increase dict size in config")
+                if source == captcha.APPSEC_SOURCE then
```

Comment:
a comment here to clarify the different behavior of captcha for APPSEC (has to be solved for every blocked req) vs REMEDIATION (has to be solved "once") would be useful for future you :)